specification.md

```diff
@@ -694,6 +695,53 @@ to `null`.
 
 ### Job
 
+#### State Model
+
+Job instances are, in this API, stateful objects.  A job's state can be
+inspected via the `job.getStatus()` method which will return a `JobStatus`
+instance on which the job's state is available as an attribute.  State
+transitions can also be received via callbacks
+
+TODO: how are callbacks registered?
```

Comment:
I believe the `setJobStatusCallback` method of the `Executor` or the `setStatusCallback` of the `Job`.

Reply:
yes, of course (*blush*)